framework/src/problems/FEProblemBase.C

```diff
@@ -180,6 +182,16 @@ FEProblemBase::validParams()
       "rz_coord_axis", rz_coord_axis, "The rotation axis (X | Y) for axisymetric coordinates");
   params.addParam<bool>(
       "kernel_coverage_check", true, "Set to false to disable kernel->subdomain coverage check");
+  params.addParam<bool>(
+      "boundary_restricted_node_integrity_check",
+      true,
+      "Set to false to disable checking of boundary restricted nodal object variable dependencies, "
+      "e.g. are the variable dependencies defined on the selected boundaries?");
+  params.addParam<bool>("boundary_restricted_elem_integrity_check",
+                        false,
```

Comment:
why false?
Too expensive?

Reply:
causes too many errors because there are side user objects and integrated boundary conditions (one example is this view-factor thingy in the head conduction module) that do wild things with non-local variable evaluations. But I was thinking I might introduce a `virtual` method on `BoundaryRestrictable` called `bool shouldCheckVariableDependencies() const` that some of these wild objects can override to return `false`. It kind of seems silly to have a sanity check that doesn't run by default ... almost no users would ever know to turn it on